test/cmpd_dtransform.c

```diff
@@ -132,25 +137,29 @@ main(void)
     free(data);
     free(data_res);
 
+    if (H5Fdelete(FILENAME, H5P_DEFAULT) < 0)
+        TEST_ERROR;
```

Comment:
May want to set any freed buffers above to NULL so a failure here doesn't double-free.

Reply:
Fixed via reordering